pkgs/build-support/dotnet/build-dotnet-module/hooks/dotnet-install-hook.sh

```diff
@@ -1,70 +1,86 @@
-# inherit arguments from derivation
-dotnetInstallFlags=( ${dotnetInstallFlags[@]-} )
-
 dotnetInstallHook() {
     echo "Executing dotnetInstallHook"
 
     runHook preInstall
 
-    if [ "${selfContainedBuild-}" ]; then
-        dotnetInstallFlags+=("--self-contained")
+    local -r hostRuntimeId=@runtimeId@
```

Comment:
There's quite substantial duplicate code here.  Any way we could reduce that?

Reply:
Are you referring to `$__structuredAttrs` branches? We can probably check for `[[ -z $__structuredAttrs ]]` instead and avoid using `*Array` variables, but we still have to re-declare arrays as local since we are appending to them in some cases, so similar LoC count.

Reply:
It's just the fact that these are duplicated in multiple hooks.  I'm not actually sure why all the hooks are separate.  Maybe we should combine them like e.g. `zig/setup-hooks.sh`.

Anyway, this is fine for now.